modular/uploader/upload_task.go

```diff
@@ -407,29 +418,29 @@ func StreamReadAt(stream io.Reader, b []byte) (int, error) {
 	}
 }
 
-func (u *UploadModular) updatePieceCheckSum(task coretask.ResumableUploadObjectTask, data []byte, isUpdate bool) error {
+func (u *UploadModular) updatePieceCheckSum(task coretask.ResumableUploadObjectTask, data []byte, isUpdate bool, dataLength uint64) error {
 	var err error
 	if isUpdate {
 		err = u.baseApp.GfSpDB().UpdateShadowPieceChecksum(task.GetObjectInfo().Id.Uint64(), piecestore.PrimarySPRedundancyIndex, hash.GenerateChecksum(data), task.GetObjectInfo().GetVersion())
 	} else {
-		err = u.baseApp.GfSpDB().UpdatePieceChecksum(task.GetObjectInfo().Id.Uint64(), piecestore.PrimarySPRedundancyIndex, hash.GenerateChecksum(data))
+		err = u.baseApp.GfSpDB().UpdatePieceChecksum(task.GetObjectInfo().Id.Uint64(), piecestore.PrimarySPRedundancyIndex, hash.GenerateChecksum(data), dataLength)
 	}
 	return err
 }
 
-func (u *UploadModular) getPieceCheckSumList(task coretask.ResumableUploadObjectTask, isUpdate bool) ([][]byte, error) {
+func (u *UploadModular) getPieceCheckSumListAndPieceSize(task coretask.ResumableUploadObjectTask, isUpdate bool) ([][]byte, uint64, error) {
```

Comment:
Can you test if the piece size can be queried correctly? dont see the value is assigned when retrieved from DB in the PR

Reply:
fixed